packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -454,31 +455,34 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                 if (this.hostPolicy.summarizerClient && trees && blobs) {
                     const blobsIdToPathMap: Map<string, string> = new Map();
                     let appCommit: string | undefined;
-                    let appTree: string | undefined;
+                    let appTreeId: string | undefined;
 
                     for (const [key, treeVal] of this.treesCache.entries()) {
-                        if (!appCommit && !appTree) {
+                        if (!appCommit && !appTreeId) {
                             for (const entry of treeVal.entries) {
                                 if (entry.path === ".app") {
                                     if (entry.type === "commit") {
                                         // This is the unacked handle of the latest summary generated.
                                         appCommit = entry.id;
                                     }
                                     if (entry.type === "tree") {
-                                        appTree = entry.id;
+                                        appTreeId = entry.id;
                                     }
                                     break;
                                 }
                             }
-                            assert(!!appCommit || !!appTree); // .app commit or tree should be first entry in first entry.
+                            assert(!!appCommit || !!appTreeId); // .app commit or tree should be first entry in first entry.
                         }
                         for (const entry of treeVal.entries) {
                             if (entry.type === "blob") {
                                 blobsIdToPathMap.set(entry.id, key === appCommit ? `/.app/${entry.path}` : `/${entry.path}`);
                             }
                         }
                     }
-
+                    assert(id !== undefined, "App tree should have a id!");
+                    const appTree = this.treesCache.get(id);
+                    assert(appTree !== undefined, "App tree should be there");
+                    this.buildHierarchyToSummaryTree(appTree);
```

Comment:
Does this call has any side effects? I.e. it returns something, but we ignore return result. What's the purpose of this call?
Should it be reflected in the name if it has some side effects?

Reply:
Maybe other way to say it - it should either return void (and be called differently), or it should not touch this.treesPathToSummaryTreeCache but return built cache and this caller should assign result of calling this function to this.treesPathToSummaryTreeCache

Reply:
Removed this piece of code as we now build the cache during getSnapshotTree call. Name now reflects what we are doing inside that function.